plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -169,22 +211,117 @@ public ConnectorPageSource createPageSource(
     {
         IcebergSplit split = (IcebergSplit) connectorSplit;
         IcebergTableHandle table = (IcebergTableHandle) connectorTable;
-
-        List<IcebergColumnHandle> icebergColumns = columns.stream()
+        Table icebergTable = table.getTable();
+        FileScanTask task = split.getTask();
+        List<IcebergColumnHandle> updateColumns = table.getUpdateColumns();
+        List<IcebergColumnHandle> queriedColumns = columns.stream()
                 .map(IcebergColumnHandle.class::cast)
                 .collect(toImmutableList());
 
         Map<Integer, Optional<String>> partitionKeys = split.getPartitionKeys();
+        Optional<StructLike> partition = task.spec().isUnpartitioned() ? Optional.empty() : Optional.of(task.file().partition());
```

Comment:
The whole `createPageSource` is hard to follow. Block of code with lots of raw index operations. Would be great to add some higher level structure to it.

Reply:
Yes I agree, I can explore that a bit more to see what's the best way to refactor this to be more readable.